src/include/optimizer/rewriter.h

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rewriter.h
+//
+// Identification: src/include/optimizer/rewriter.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <memory>
+
+#include "expression/abstract_expression.h"
+#include "optimizer/optimizer_metadata.h"
+#include "optimizer/optimizer_task_pool.h"
+#include "optimizer/absexpr_expression.h"
+
+namespace peloton {
+namespace optimizer {
+
+class Rewriter {
+
+ public:
+  Rewriter();
+  void Reset();
+
+  DISALLOW_COPY_AND_MOVE(Rewriter);
+
+  OptimizerMetadata &GetMetadata() { return metadata_; }
```

Comment:
Doxygen will complain about a lack of doxygen documentation on public functions.

Reply:
Resolved.